src/main/java/org/wildfly/security/auth/server/RealmIdentity.java

```diff
@@ -118,6 +119,29 @@ default Principal getRealmIdentityPrincipal() {
     boolean verifyEvidence(Evidence evidence) throws RealmUnavailableException;
 
     /**
+     * Determine if the given evidence can be trusted in lieu of verifying or acquiring a credential. This method is
+     * intended to be used for identity propagation purposes.
+     *
+     * @param evidence the evidence to check (must not be {@code null})
+     * @return {@code true} if the given evidence can be trusted, {@code false} otherwise
+     * @throws RealmUnavailableException if the realm is not able to handle requests for any reason
+     */
+    default boolean checkEvidenceTrusted(Evidence evidence) throws RealmUnavailableException {
```

Comment:
We had discussed delegating the trust check to the realm level to avoid having to introduce a special case for SecurityIdentityEvidence in ServerAuthenticationContext#verifyEvidence.